src/main/java/io/cdap/plugin/gcp/bigquery/sink/AvroRecordWriter.java

```diff
@@ -114,4 +118,20 @@ public long sync() throws IOException {
     }
     return 0;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof AvroRecordWriter)) return false;
+    AvroRecordWriter that = (AvroRecordWriter) o;
+
+    return syncInterval == that.syncInterval && Objects.equals(mAvroFileWriter, that.mAvroFileWriter)
```

Comment:
@tivv 
Did not include supplier as I could not find a way to compare 2 `Supplier` .

Is there any way to do so ?

Reply:
Why do you need it for unit test? Usually adding equals not to a data object is not a good idea as it has such challenges. You can necessary properties for testing and use org.hamcrest.beans.SamePropertyValuesAs

Reply:
This also has a benefit of good description instead of "not equal, sorry", especially without good toString

Reply:
This is for unit test for `AvroOutputFormat` , for method `getRecordWriter` which would return a `AvroRecordWriter` 

hence I wanted to have a equals method to compare Actual & expected object.

Reply:
For non-DTOs you often can't compare those directly as they don't need (and often can't) implement equals. You should decide which specific properties to compare, provide getters (may be with "just for test" visibility) and compare those properties. The classes referecned above help you to do the comparison

Reply:
Thanks. 
Implemented the same.